dubbo-common/src/main/java/org/apache/dubbo/common/utils/NetUtils.java

```diff
@@ -196,7 +196,7 @@ public static InetSocketAddress getLocalSocketAddress(String host, int port) {
     }
 
     static boolean isValidV4Address(InetAddress address) {
-        if (address == null || address.isLoopbackAddress()) {
+        if (address == null || address.isLoopbackAddress() || address.isLinkLocalAddress()) {
```

Comment:
add a para allowLinkLocal in some related method and user config the value in system config file which it's default value if true to prevent the situation that there is no any non-local interface,  it's ok?

Reply:
i've done the work in local,if my idea is available and need my code ,i 'll push a request under this issue!

Reply:
Refer to https://github.com/apache/dubbo/pull/14793/files#r1814182854, `address.isLinkLocalAddress()` shouldn't affect if there is no non-local address. I don't have a Dubbo environment locally, Can you help test the changes? thanks a lot.